### PR TITLE
Add size property

### PR DESCRIPTION
**Pull request by rehkmansa**
Allows passing `numbers` or `small` (16px) | `medium` (default, 24px) | `large` (32px)

### CODE REVIEW

File: src/components/icon/icon.tsx [hunk 1]
```diff
@@ -1,21 +1,43 @@
...
+  small: 16,
+  medium: 24,
+  large: 32,
+};
+
+const assertSize = (size: IconSize | number): number => {
+  if (typeof size === "number") return size;
+
+  return ICON_SIZE[size];
+};
```
Comment (ant0n7) [inline src/components/icon/icon.tsx]: "assert" might be wrong as this is kind of a getter

File: scripts/create-icon.ts [hunk 2]
```diff
@@ -40,22 +40,29 @@ const createElement = async (name: string, node: unknown) => {
 
   const jsxString = await svgToJsx(node as string);
 
+  const size = (key: string) => `${key} ?? size`;
+
   const updatedJsxString = jsxString.replace(
     /<svg([^>]*)>/,
-    `<svg$1 {...props} ref={ref}>`,
+    `<svg$1 width={${size("width")}} height={${size("height")}} ref={ref} {...rest}>`,
```
Comment (ant0n7) [inline scripts/create-icon.ts]: can you explain this please?

File: src/components/icon/types.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import type { SVGAttributes } from "react";
+
+export type IconSize = "small" | "large" | "medium";
+
+export interface DefaultIconProps extends SVGAttributes<SVGElement> {
+  size?: number;
```
Comment (ant0n7) [inline src/components/icon/types.ts:6]: i thought size can be a number or IconSize. why is this here only a number?

why is this props interface outside of the component?

File: scripts/create-icon.ts (hunk 2 above)
Comment (rehkmansa) [reply scripts/create-icon.ts]: There is the options to pass width or height, if not passed use size as the default value

File: src/components/icon/types.ts (hunk 3 above)
Comment (rehkmansa) [reply src/components/icon/types.ts:6]: Its used by two different components,

Icon (is the wrapper that resolves to the individual icon,  converts the size to a number)
The individual icons accept size as number only, no need the run the getter here when Icon already exists
